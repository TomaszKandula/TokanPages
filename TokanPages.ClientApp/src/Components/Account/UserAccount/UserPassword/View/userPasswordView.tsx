import * as React from "react";
import Box from "@material-ui/core/Box";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Container from "@material-ui/core/Container";
import Skeleton from "@material-ui/lab/Skeleton";
import { ViewProperties } from "../../../../../Shared/interfaces";
import { UserPasswordStyle } from "./userPasswordStyle";

import { 
    Button, 
    CircularProgress, 
    Divider, 
    Grid, 
    TextField, 
    Typography
} from "@material-ui/core";

import { 
    SectionAccessDenied, 
    SectionAccountPassword
} from "../../../../../Api/Models";

interface Properties extends ViewProperties
{    
    oldPassword: string;
    newPassword: string;
    confirmPassword: string;
    passwordFormProgress: boolean;
    passwordKeyHandler: any;
    passwordFormHandler: any;
    passwordButtonHandler: any;
    sectionAccessDenied: SectionAccessDenied;
    sectionAccountPassword: SectionAccountPassword;
}

const UpdatePasswordButton = (props: Properties): JSX.Element => 
{
    const classes = UserPasswordStyle();
    return(
        <Button 
            fullWidth 
            type="submit" 
            variant="contained" 
            onClick={props.passwordButtonHandler} 
            disabled={props.passwordFormProgress} 
            className={classes.button_update}>
            {!props.passwordFormProgress 
            ? props.sectionAccountPassword?.updateButtonText 
            : <CircularProgress size={20} />}
        </Button>
    );
}

const CustomDivider = (args: { marginTop: number, marginBottom: number }) => 
{
    const classes = UserPasswordStyle();
    return(
        <Box mt={args.marginTop} mb={args.marginBottom}>
            <Divider className={classes.divider} />
        </Box>
    );
}

export const UserPasswordView = (props: Properties): JSX.Element => 
{
    const classes = UserPasswordStyle();
    return(
        <section className={classes.section}>
            <Container maxWidth="md">                
                <Box pt={5} pb={5}>
                    <Card elevation={0} className={classes.card}>
                        <CardContent className={classes.card_content}>
                            <Box pt={0} pb={0}>
                                <Typography component="span" className={classes.caption}>
                                    {props.isLoading 
                                    ? <Skeleton variant="text" /> 
                                    : props.sectionAccountPassword?.caption}
                                </Typography>
                            </Box>
                            <CustomDivider marginTop={2} marginBottom={1} />
                            <Box pt={5} pb={1}>
                                <Grid container spacing={2}>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            {props.isLoading 
                                            ? <Skeleton variant="text" /> 
                                            : props.sectionAccountPassword?.labelOldPassword}
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="oldPassword" 
                                            name="oldPassword" 
                                            variant="outlined" 
                                            type="password"
                                            value={props.oldPassword}
                                            onKeyUp={props.passwordKeyHandler}
                                            onChange={props.passwordFormHandler} 
                                        />}
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            {props.isLoading 
                                            ? <Skeleton variant="text" /> 
                                            : props.sectionAccountPassword?.labelNewPassword}
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="newPassword" 
                                            name="newPassword" 
                                            variant="outlined" 
                                            type="password"
                                            value={props.newPassword}
                                            onKeyUp={props.passwordKeyHandler}
                                            onChange={props.passwordFormHandler} 
                                        />}
                                    </Grid>
                                    <Grid item xs={12} sm={3}>
                                        <Typography component="span" className={classes.label}>
                                            {props.isLoading 
                                            ? <Skeleton variant="text" /> 
                                            : props.sectionAccountPassword?.labelConfirmPassword}
                                        </Typography>
                                    </Grid>
                                    <Grid item xs={12} sm={9}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="100%" height="40px" />
                                        : <TextField 
                                            required 
                                            fullWidth 
                                            id="confirmPassword" 
                                            name="confirmPassword" 
                                            variant="outlined" 
                                            type="password"
                                            value={props.confirmPassword}
                                            onKeyUp={props.passwordKeyHandler}
                                            onChange={props.passwordFormHandler} 
                                        />}
                                    </Grid>
                                </Grid>
                                <CustomDivider marginTop={5} marginBottom={2} />
                                <Grid className={classes.button_container_update}>
                                    <Box my={2}>
                                        {props.isLoading 
                                        ? <Skeleton variant="rect" width="150px" height="40px" /> 
                                        : <UpdatePasswordButton {...props} />}
                                    </Box>
                                </Grid>
                            </Box>
                        </CardContent>
                    </Card>
                </Box>
            </Container>
        </section>
    );
}
