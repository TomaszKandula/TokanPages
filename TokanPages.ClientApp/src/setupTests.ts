// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
