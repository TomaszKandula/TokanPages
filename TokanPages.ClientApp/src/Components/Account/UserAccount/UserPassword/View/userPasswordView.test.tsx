import "../../../../../setupTests";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";
import { UserPasswordView } from "./userPasswordView";

describe("test account group component: userPasswordView", () => {
    it("should render correctly '<UserPasswordView />' when content is loaded.", () => {
        const html = render(
            <BrowserRouter>
                <UserPasswordView
                    isLoading={false}
                    isMobile={false}
                    oldPassword="Provide old password"
                    newPassword="Provide new password"
                    confirmPassword="Confirm new password"
                    keyHandler={jest.fn()}
                    formProgress={false}
                    formHandler={jest.fn()}
                    buttonHandler={jest.fn()}
                    sectionAccountPassword={{
                        caption: "Change Password",
                        labelOldPassword: "123456987",
                        labelNewPassword: "ester_exposito_2020",
                        labelConfirmPassword: "ester_exposito_2020",
                        updateButtonText: "Update",
                    }}
                />
            </BrowserRouter>
        );

        expect(html).toMatchSnapshot();
    });
});
