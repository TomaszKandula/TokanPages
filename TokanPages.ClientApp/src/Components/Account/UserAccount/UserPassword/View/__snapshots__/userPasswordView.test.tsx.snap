// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test account group component: userPasswordView should render correctly '<UserPasswordView />' when content is loaded. 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="section "
      >
        <div
          class="MuiContainer-root container-wide MuiContainer-maxWidthLg"
        >
          <div
            class="pb-40"
          >
            <div
              class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root card-content"
              >
                <span
                  class="MuiTypography-root caption black MuiTypography-body1"
                >
                  Change Password
                </span>
                <div
                  class="mt-15 mb-8"
                >
                  <hr
                    class="MuiDivider-root divider"
                  />
                </div>
                <div
                  class="pt-40 pb-8"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  >
                    <div
                      class="MuiGrid-root label-centered MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
                    >
                      <span
                        class="MuiTypography-root label MuiTypography-body1"
                      >
                        123456987
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                        style="margin: 0px;"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                          data-shrink="true"
                          for="oldPassword"
                        >
                          
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                            id="oldPassword"
                            name="oldPassword"
                            type="password"
                            value="Provide old password"
                          />
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                          >
                            <button
                              aria-label="toggle password visibility"
                              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                              tabindex="0"
                              type="button"
                            >
                              <span
                                class="MuiIconButton-label"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                  />
                                </svg>
                              </span>
                              <span
                                class="MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                            style="padding-left: 8px;"
                          >
                            <legend
                              class="PrivateNotchedOutline-legend-2"
                              style="width: 60.5px;"
                            >
                              <span>
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root label-centered MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
                    >
                      <span
                        class="MuiTypography-root label MuiTypography-body1"
                      >
                        ester_exposito_2020
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                        style="margin: 0px;"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                          data-shrink="true"
                          for="newPassword"
                        >
                          
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                            id="newPassword"
                            name="newPassword"
                            type="password"
                            value="Provide new password"
                          />
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                          >
                            <button
                              aria-label="toggle password visibility"
                              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                              tabindex="0"
                              type="button"
                            >
                              <span
                                class="MuiIconButton-label"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                  />
                                </svg>
                              </span>
                            </button>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                            style="padding-left: 8px;"
                          >
                            <legend
                              class="PrivateNotchedOutline-legend-6"
                              style="width: 60.5px;"
                            >
                              <span>
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root label-centered MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
                    >
                      <span
                        class="MuiTypography-root label MuiTypography-body1"
                      >
                        ester_exposito_2020
                      </span>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
                    >
                      <div
                        class="MuiFormControl-root MuiFormControl-fullWidth"
                        style="margin: 0px;"
                      >
                        <label
                          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                          data-shrink="true"
                          for="confirmPassword"
                        >
                          
                        </label>
                        <div
                          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                            id="confirmPassword"
                            name="confirmPassword"
                            type="password"
                            value="Confirm new password"
                          />
                          <div
                            class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                          >
                            <button
                              aria-label="toggle password visibility"
                              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                              tabindex="0"
                              type="button"
                            >
                              <span
                                class="MuiIconButton-label"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                  />
                                </svg>
                              </span>
                            </button>
                          </div>
                          <fieldset
                            aria-hidden="true"
                            class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                            style="padding-left: 8px;"
                          >
                            <legend
                              class="PrivateNotchedOutline-legend-10"
                              style="width: 60.5px;"
                            >
                              <span>
                                ​
                              </span>
                            </legend>
                          </fieldset>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="mt-40 mb-15"
                  >
                    <hr
                      class="MuiDivider-root divider"
                    />
                  </div>
                  <div
                    class="MuiGrid-root button-container-update"
                  >
                    <div
                      class="mt-15 mb-15"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained button-update MuiButton-fullWidth"
                        tabindex="0"
                        type="submit"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          Update
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="section "
    >
      <div
        class="MuiContainer-root container-wide MuiContainer-maxWidthLg"
      >
        <div
          class="pb-40"
        >
          <div
            class="MuiPaper-root MuiCard-root card MuiPaper-elevation0 MuiPaper-rounded"
          >
            <div
              class="MuiCardContent-root card-content"
            >
              <span
                class="MuiTypography-root caption black MuiTypography-body1"
              >
                Change Password
              </span>
              <div
                class="mt-15 mb-8"
              >
                <hr
                  class="MuiDivider-root divider"
                />
              </div>
              <div
                class="pt-40 pb-8"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                >
                  <div
                    class="MuiGrid-root label-centered MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
                  >
                    <span
                      class="MuiTypography-root label MuiTypography-body1"
                    >
                      123456987
                    </span>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth"
                      style="margin: 0px;"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="oldPassword"
                      >
                        
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                          id="oldPassword"
                          name="oldPassword"
                          type="password"
                          value="Provide old password"
                        />
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                        >
                          <button
                            aria-label="toggle password visibility"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                />
                              </svg>
                            </span>
                            <span
                              class="MuiTouchRipple-root"
                            />
                          </button>
                        </div>
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-2"
                            style="width: 60.5px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root label-centered MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
                  >
                    <span
                      class="MuiTypography-root label MuiTypography-body1"
                    >
                      ester_exposito_2020
                    </span>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth"
                      style="margin: 0px;"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="newPassword"
                      >
                        
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                          id="newPassword"
                          name="newPassword"
                          type="password"
                          value="Provide new password"
                        />
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                        >
                          <button
                            aria-label="toggle password visibility"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-6"
                            style="width: 60.5px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root label-centered MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
                  >
                    <span
                      class="MuiTypography-root label MuiTypography-body1"
                    >
                      ester_exposito_2020
                    </span>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-9"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth"
                      style="margin: 0px;"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
                        data-shrink="true"
                        for="confirmPassword"
                      >
                        
                      </label>
                      <div
                        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                          id="confirmPassword"
                          name="confirmPassword"
                          type="password"
                          value="Confirm new password"
                        />
                        <div
                          class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                        >
                          <button
                            aria-label="toggle password visibility"
                            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="MuiIconButton-label"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                                />
                              </svg>
                            </span>
                          </button>
                        </div>
                        <fieldset
                          aria-hidden="true"
                          class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
                          style="padding-left: 8px;"
                        >
                          <legend
                            class="PrivateNotchedOutline-legend-10"
                            style="width: 60.5px;"
                          >
                            <span>
                              ​
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="mt-40 mb-15"
                >
                  <hr
                    class="MuiDivider-root divider"
                  />
                </div>
                <div
                  class="MuiGrid-root button-container-update"
                >
                  <div
                    class="mt-15 mb-15"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained button-update MuiButton-fullWidth"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Update
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
